### PR TITLE
Add first version of a CPG validator.

### PR DESCRIPTION
**Pull request by ml86**
This version only checks the relations between nodes and not
the node properties itself.

### CODE REVIEW

File: cpgvalidator/src/main/scala/io/shiftleft/cpgvalidator/CpgValidator.scala [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+    validateInFacts()
+
+    errorRegistry.getErrorCount == 0
+  }
+
+  def logValidationErrors(): Unit = {
+    errorRegistry.logValidationErrors()
+  }
+
+  private def validateOutFacts(): Unit = {
```
Comment (fabsx00) [inline cpgvalidator/src/main/scala/io/shiftleft/cpgvalidator/CpgValidator.scala]: `validateOutFacts` and `validateInFacts` look pretty similar. Any chance we can reduce some code duplication here?

File: cpgvalidator/src/main/scala/io/shiftleft/cpgvalidator/CpgValidator.scala [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+
+            val allowedEdgeTypes = inFactsByEdgeType.map(_._1)
+            validateAllInEdgesTypes(dstNode, inEdges, allowedEdgeTypes)
+
+          }
+          .iterate()
+    }
+  }
+
+  private def validateOutDegree(srcNode: Vertex, actualDstNodes: List[Vertex], outFact: OutFact): Unit = {
```
Comment (fabsx00) [inline cpgvalidator/src/main/scala/io/shiftleft/cpgvalidator/CpgValidator.scala]: Same question for *Degree methods.

File: cpgvalidator/src/main/scala/io/shiftleft/cpgvalidator/CpgValidator.scala (hunk 1 above)
Comment (ml86) [reply cpgvalidator/src/main/scala/io/shiftleft/cpgvalidator/CpgValidator.scala]: Yes can be deduplicated. So far i had no time to do it.

File: cpgvalidator/src/main/scala/io/shiftleft/cpgvalidator/CpgValidator.scala (hunk 2 above)
Comment (ml86) [reply cpgvalidator/src/main/scala/io/shiftleft/cpgvalidator/CpgValidator.scala]: See above.

**Pull request merged**